### PR TITLE
[JENKINS-28013,JENKINS-28012] - Revert changes for JENKINS-10629

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Thanks to @KostyaSha, we were able to create a 100%-reproducible test case for JENKINS-28013. The issue requires additional investigation and much more unit tests, so I propose to revert changes for JENKINS-10629 (excluding additional unit-tests) in order to release a stable core

@reviewbybees

### CODE REVIEW

**Comment by jtnord** [comment]
:+1: assuming this is a revert - which I have not done a full comparison on :)

BTW: would be good if the reproducable test-case is documented in the issue tracker :-)

**Comment by jglick** [comment]
Well it leaves the tests in place; the crucial one was already `@Ignore`d. So :+1:

**Comment by KostyaSha** [comment]
:+1: for revert
After debugging i can say that issue maybe in any place. Either input/output, either inflater/deflater, either tar parameters produces corrupted output stream that becomes a really "truncated archive". Commons-compress has many bugs and jzlib is not supported at all. Hudson classes duplicates commons and produces spaghetti. Issue happens near "x * 512" bytes border.  Here many things that can be tried:
- disable compression
- review streams and their buffering
- isolate pipe from channel
- review tar class wrappers and tar configuration

File: core/src/main/java/hudson/org/apache/tools/tar/TarOutputStream.java [hunk 1]
```diff
@@ -35,11 +35,8 @@
  * The TarOutputStream writes a UNIX tar archive as an OutputStream.
  * Methods are provided to put entries, and then write their contents
  * by writing to this stream using write().
- * 
- * @deprecated Use {@link org.apache.commons.compress.archivers.tar.TarArchiveOutputStream} instead
```
Comment (KostyaSha) [inline core/src/main/java/hudson/org/apache/tools/tar/TarOutputStream.java]: You need introduce @until :laughing:

**Pull request merged**